internal/lifecycle/readme.go

```diff
@@ -79,17 +79,19 @@ func (cb codeBlock) toCommands(serviceName, gcrURL string) ([]*exec.Cmd, error)
 			line = line + l
 		}
 
-		line = os.ExpandEnv(line)
 		line = gcrURLRegexp.ReplaceAllString(line, gcrURL)
 		line = replaceServiceName(line, serviceName)
-		sp := strings.Split(line, " ")
 
 		var cmd *exec.Cmd
-		if sp[0] == "gcloud" {
-			a := append(util.GcloudCommonFlags, sp[1:]...)
-			cmd = exec.Command("gcloud", a...)
+		if gcloudCommandRegexp.MatchString(line) {
+			// Splitting command to add util.GcloudCommonFlags to front of command.
+			// TODO: Ideally want to use a library that splits commands according to bash specification.
+			sp := strings.Split(line, " ")
```

Comment:
Could you just split once (after `gcloud`) using `SplitN`? I'm concerned that this will alter whitespace later in the command, which could be an issue (e.g. for whitespace in a quoted arg).

Reply:
I strongly recommend avoiding the game of CLI arg manipulation. Most gcloud flags are also available with config and environment variable overrides. I think the latter would work nicely here.

Override the environment for the `exec.Command` operation by combining the existing environment variables with the items you need to customize.

The Go technique: https://pkg.go.dev/os/exec?tab=doc#example-Command-Environment
For `--quiet`, the documented variable is `CLOUDSDK_CORE_DISABLE_PROMPTS`.

Reply:
Resolved, thank you both!